### ISSUE TITLE
Tackling Issue #15

### ISSUE THREAD

**Issue by QiyaoWei** [mosaicml/composer#857]
## New Method

Hi! This is Qiyao Wei, an applicant to Mosaic ML. I thought I would start my contribution by tackling one of the "good first issues", specifically #15. It makes sense that we want a `predict()` method, but are there specifications for what the input and output for this method is? For example, do we expect the user to input a dataloader or just a batch of data? Are we outputting the softmax logit values or one of the ten classification classes? I have attached an implementation below, and please bear with me as I am getting familiar with the Trainer interface! I would much appreciate comments both to my questions and my code!

## Motivation

Adding `predict()` expands user functionality for`Trainer`

## Attribution

 
This issue is wrt #15

## [Optional] Implementation

```
    def predict(self, loader, batchsize):

        self.state.model.eval()
        for batch_idx, self.state.batch in enumerate(
            itertools.islice(loader, batchsize)):

            self.state.batch = self._device.batch_to_device(self.state.batch)
            if self.deepspeed_enabled:
                self.state.batch = _fix_batch_precision_for_deepspeed(self.state.batch, self.state.precision)

            with torch.no_grad():
                outputs, targets = self._original_model.validate(self.state.batch)
        return outputs, targets
```
